apis/Google.Cloud.Firestore/Google.Cloud.Firestore/AggregateQuerySnapshot.cs

```diff
@@ -0,0 +1,60 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at 
+//
+// https://www.apache.org/licenses/LICENSE-2.0 
+//
+// Unless required by applicable law or agreed to in writing, software 
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and 
+// limitations under the License.
+
+using Google.Api.Gax;
+using System;
+
+namespace Google.Cloud.Firestore;
+
+/// <summary>
+/// An immutable snapshot of aggregate query results.
+/// </summary>
+public sealed class AggregateQuerySnapshot : IEquatable<AggregateQuerySnapshot>
+{
+    /// <summary>
+    /// The query producing this snapshot.
+    /// </summary>
+    public AggregateQuery Query { get; }
+
+    /// <summary>
+    /// The time at which the snapshot was read.
+    /// </summary>
+    public Timestamp ReadTime { get; }
+
+    /// <summary>
+    /// Number of documents that matches the query.

```

Comment:
Possibly add a note about limits?
What will this return in the future when we support non-count aggregations, and a count hasn't been specified?
(Perhaps the return type should actually be `long?` instead?) I'm genuinely not sure.

Reply:
Thanks for your feedback @jskeet .Will update it and let you know.

Reply:
Yes Jon I will add a note about the limit thing. On return thing: When we support non-count aggregations and count hasn't been specified in that case the value of count would be "null". I have updated the return type to be long?.
Thanks